apache/superset
# fix(assets api): import replaces dashboard

### SUMMARY

For existing dashboards, modifications imported via `/api/v1/assets/import/` are only appended if the modified dashboard contains a new chart. Modifications to dashboards that remove a chart are not implemented when imported through the assets API.

This change updates the assets API import method to remove all existing references to chart ids for the imported dashboard(s) before inserting the new chart ids. The change should benefit the management of Superset assets under source control.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


### TESTING INSTRUCTIONS

Remove a chart from a dashboard:
- Export a dashboard and unzip assets bundle.
- Remove a chart and all references to it from the exported yaml files
- Import modified assets bundle as a zip by posting to `/api/v1/assets/import/` (see https://superset.apache.org/docs/api/#api)
- Navigate to the modified dashboard in the Superset UI and observe the deleted chart is no longer visible

Add a chart to a dashboard:
- In the same dashboard, remove a chart using the UI and save the dashboard
- Import the previously exported assets bundle as a zip by posting to `/api/v1/assets/import/` 
- Navigate to the modified dashboard in the Superset UI and observe the deleted chart is now reinstated


### ADDITIONAL INFORMATION


- [x] Has associated issue: https://github.com/apache/superset/issues/22127
